src/ExportMap.js

```diff
@@ -781,12 +782,29 @@ export function recursivePatternCapture(pattern, callback) {
   }
 }
 
+let parserOptionsHash = '';
+let prevParserOptions = null;
+let settingsHash = '';
+let prevSettings = null;
 /**
  * don't hold full context object in memory, just grab what we need.
+ * also calculate a cashKey, where parts of the cacheKey hash are memoized
  */
 function childContext(path, context) {
   const { settings, parserOptions, parserPath } = context;
+
+  if (!deepEqual(settings, prevSettings)){
+    settingsHash = hashObject({ settings }).digest('hex');
+    prevSettings = settings;
+  }
+
+  if (!deepEqual(parserOptions, prevParserOptions)){
+    parserOptionsHash = hashObject({ parserOptions }).digest('hex');
+    prevParserOptions = parserOptions;
+  }
+
   return {
+    cacheKey: String(parserPath) + parserOptionsHash + settingsHash + String(path),
```

Comment:
Wasn't too sure, whether we can rely on `parserPath` and `path` being strings.

Reply:
I think we can, but it's fine to stringify them just in case.